src/jetton/JettonMaster.ts

```diff
@@ -9,35 +9,38 @@
 import { Address, beginCell, Contract, ContractProvider } from "@ton/core";
 
 export class JettonMaster implements Contract {
+  static create(address: Address) {
+    return new JettonMaster(address);
+  }
 
-    static create(address: Address) {
-        return new JettonMaster(address);
-    }
-
-    readonly address: Address;
-
-    constructor(address: Address) {
-        this.address = address;
-    }
-
-    async getWalletAddress(provider: ContractProvider, owner: Address) {
-        let res = await provider.get('get_wallet_address', [{ type: 'slice', cell: beginCell().storeAddress(owner).endCell() }]);
-        return res.stack.readAddress();
-    }
-
-    async getJettonData(provider: ContractProvider) {
-        let res = await provider.get('get_jetton_data', []);
-        let totalSupply = res.stack.readBigNumber();
-        let mintable = res.stack.readBoolean();
-        let adminAddress = res.stack.readAddress();
-        let content = res.stack.readCell();
-        let walletCode = res.stack.readCell();
-        return {
-            totalSupply,
-            mintable,
-            adminAddress,
-            content,
-            walletCode
-        };
-    }
-}
\ No newline at end of file
+  readonly address: Address;
+
+  constructor(address: Address) {
+    this.address = address;
+  }
+
+  async getWalletAddress(provider: ContractProvider, owner: Address) {
+    let res = await provider.get("get_wallet_address", [
+      { type: "slice", cell: beginCell().storeAddress(owner).endCell() },
+    ]);
+    return res.stack.readAddress();
+  }
+
+  async getJettonData(provider: ContractProvider) {
+    let res = await provider.get("get_jetton_data", []);
+
+    const totalSupply = res.stack.readBigNumber();
+    const mintable = res.stack.readBoolean();
+    const content = res.stack.readCell();
+    const walletCode = res.stack.readCell();
+    const adminAddress = res.stack.readAddressOpt();
```

Comment:
sorry, since the structure of jetton master is
```
total_supply: int
mintable: boolean
admin_address: address
jetton_content: cell
jetton_wallet_code: cell
```

[reference](https://docs.ton.org/develop/dapps/asset-processing/jettons#:~:text=The%20jetton%20master%20smart%20contract,%2C%20or%20the%20metadata%20itself.&text=It%20is%20possible%20for%20any,nearly%20identical%20to%20the%20original.)

you need to read the stack sequentially.
```typescript
const totalSupply = res.stack.readBigNumber();
const mintable = res.stack.readBoolean();
const adminAddress = res.stack.readAddressOpt();
const content = res.stack.readCell();
const walletCode = res.stack.readCell();
```

Reply:
fixed